dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -1112,6 +1112,150 @@ def median_abs_deviation(self):
                                    [edge_prev, edge_cur])
         return median_abs_dev
 
+    @property
+    def median_abs_deviation(self):
+        """
+        Get median absolute deviation.
+
+        :return: median absolute deviation
+        """
+        if not self._has_histogram:
+            return np.nan
+
+        bin_counts = self._stored_histogram['histogram']['bin_counts']
+        bin_edges = self._stored_histogram['histogram']['bin_edges']
+
+        # normalize bin counts
+        bin_counts = bin_counts.astype(float)
+        normalized_bin_counts = bin_counts / np.sum(bin_counts)
+
+        # calculate deviations from median
+        median = self.quantiles[499]
+        bin_edges = bin_edges - median
+
+        # find the break point to fold the deviation list
+        id_zero = None
+        for i in range(len(bin_edges)):
+            if bin_edges[i] > 0:
+                id_zero = i
+                break
+
+        # if all bin edges are positive or negative (no break point),
+        # the median value is actually 0
+        if id_zero is None or id_zero == 1:
+            return 0
+
+        # otherwise, generate two folds of deviation
+        bin_counts_pos = np.append(
+            [normalized_bin_counts[id_zero - 1] * (bin_edges[id_zero] /
+                (bin_edges[id_zero] - bin_edges[id_zero - 1]))],
+            normalized_bin_counts[id_zero:])
+        bin_edges_pos = np.append([0], bin_edges[id_zero:])
+
+        bin_counts_neg = np.append([normalized_bin_counts[id_zero - 1] -
+             normalized_bin_counts[id_zero - 1] * (bin_edges[id_zero] /
+                (bin_edges[id_zero] - bin_edges[id_zero - 1]))],
+            normalized_bin_counts[:id_zero - 1][::-1])
+        bin_edges_neg = np.append([0], -bin_edges[:id_zero][::-1])
```

Comment:
this is where you split to pos and negative, but also more importantly, split the bin if it is within a bin. I'm curious of the case when the bin occurs directly at a bin_edge.